### PR TITLE
Reformat and doc

### PR DESCRIPTION
**Pull request by MatBarba**
Makes it pass `mypy` and `pylint` for cleaner dev down the line

### CODE REVIEW

File: src/veupath/redmine/client/__init__.py [hunk 1]
```diff
@@ -44,10 +46,12 @@ def _map_key(self, name: str) -> str:
         return key
 
     def set_field(self, field_name, field_value):
+        """Set a field value."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/__init__.py:49]: For a future implementation: this and the following method could have been done via `__set__` and `__get__` methods, making the syntax "more readable" when using this class.

File: src/veupath/redmine/client/checks/__init__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!env python3
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/checks/__init__.py]:
```suggestion

```

File: src/veupath/redmine/client/checks/check_genome_issues.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""Check, parse and store genome Redmine issues from VeupathDB."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/checks/check_genome_issues.py]:
```suggestion
"""Check, parse and store genome Redmine issues from VEuPathDB."""
```

File: src/veupath/redmine/client/veupath_params.py [hunk 4]
```diff
@@ -13,9 +13,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""VeupathDB Redmine server parameters in one object."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/veupath_params.py]:
```suggestion
"""VEuPathDB Redmine server parameters in one object."""
```

File: src/veupath/redmine/client/veupath_params.py [hunk 5]
```diff
@@ -13,9 +13,12 @@
...
 
 
 class VeupathParams:
+    """Store basic parameters for the VeupathDB Redmine server."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/veupath_params.py]:
```suggestion
    """Store basic parameters for the VEuPathDB Redmine server."""
```

File: src/veupath/redmine/client/checks/check_organism_abbrevs.py [hunk 6]
```diff
@@ -44,7 +46,21 @@ def get_genome_issues(redmine: VeupathRedmineClient) -> list:
     return genomes
 
 
-def categorize_abbrevs(issues: List[RedmineIssue], cur_abbrevs_path: str = "") -> Dict[str, List[Genome]]:
+def categorize_abbrevs(
+    issues: List[RedmineIssue], cur_abbrevs_path: Union[PathLike, None] = None
+) -> Dict[str, List[Genome]]:
+    """Returns a dict of lists if genome issues depending on their categorization.
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/checks/check_organism_abbrevs.py]:
```suggestion
    """Returns a dict of lists of genome issues depending on their categorization.
```

File: src/veupath/redmine/client/checks/check_rnaseq_issues.py [hunk 7]
```diff
@@ -13,15 +13,16 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""Check, parse and store RNA-Seq Redmine issues from VeupathDB."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/checks/check_rnaseq_issues.py]:
```suggestion
"""Check, parse and store RNA-Seq Redmine issues from VEuPathDB."""
```

File: src/veupath/redmine/client/genome.py [hunk 8]
```diff
@@ -208,6 +203,7 @@ def _get_gff(self) -> None:
             self.gff = gff_path
 
     def _get_replacement(self) -> None:
+        """Set `is_replacement` to true if the genome is expected to be a replacement."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/genome.py]:
```suggestion
        """Set `is_replacement` to True if the genome is expected to be a replacement."""
```
I tend to like to set boolean values in capital case to follow Python coding in the docstrings, I find these more readable.

File: src/veupath/redmine/client/genome.py [hunk 9]
```diff
@@ -243,24 +240,23 @@ def _get_insdc_metadata(self) -> Dict[str, Any]:
                         break
         return summary
 
-    def get_assembly_metadata(self, accession):
+    def get_assembly_metadata(self, accession: str):
+        """Returns a metadata summary from entrez for the provided accession."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/genome.py]:
```suggestion
        """Returns a metadata summary from Entrez for the provided accession."""
```

File: src/veupath/redmine/client/genome.py [hunk 10]
```diff
@@ -243,24 +240,23 @@ def _get_insdc_metadata(self) -> Dict[str, Any]:
...
             return True
-        else:
-            return False
+        return False
 
     def _check_datatype(self) -> None:
-        """
-        Check if we expect an annotation with a gff from INSDC/GFF2Load
-        """
+        """Checks if we expect an annotation with a gff from INSDC/GFF2Load."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/genome.py]:
```suggestion
        """Checks if we expect an annotation with a GFF from INSDC/GFF2Load."""
```

File: src/veupath/redmine/client/redmine_issue.py [hunk 11]
```diff
@@ -48,20 +49,25 @@ def __init__(self, issue):
...
     def disable_log(self) -> None:
+        """Disable log message storage."""
         self.do_log = False
 
     def enable_log(self) -> None:
+        """Activate log message storage."""
         self.do_log = True
 
     def _get_component(self) -> str:
+        """Get the VeupathDB component DB from the issue."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/redmine_issue.py]:
```suggestion
        """Get the VEuPathDB component DB from the issue."""
```

File: src/veupath/redmine/client/redmine_issue.py [hunk 12]
```diff
@@ -78,6 +84,7 @@ def _get_component(self) -> str:
         return component
 
     def _get_team(self) -> str:
+        """Get the VeupathDB team from the issue."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/redmine_issue.py]:
```suggestion
        """Get the VEuPathDB team from the issue."""
```

File: src/veupath/redmine/client/redmine_issue.py [hunk 13]
```diff
@@ -112,22 +121,26 @@ def _get_experimental_organism(self) -> str:
...
 
     def _get_datatype(self) -> str:
+        """Get the datatype from the issue."""
         try:
             return self.custom["DataType"]
         except KeyError:
             return ""
 
     def redmine_link(self) -> str:
+        """Create an html link string to the redmine ticket."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/redmine_issue.py]:
```suggestion
        """Create an HTML link string to the Redmine ticket."""
```

File: src/veupath/redmine/client/redmine_issue.py [hunk 14]
```diff
@@ -112,22 +121,26 @@ def _get_experimental_organism(self) -> str:
...
 
     def _get_operations(self) -> List[str]:
+        """Get the list of EBI operations from the issue."""
         try:
             return self.custom["EBI operations"]
         except KeyError:
             return []
 
     def _get_build(self) -> str:
+        """Get the VeupathDB build from the issue."""
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/redmine_issue.py]:
```suggestion
        """Get the VEuPathDB build from the issue."""
```

File: src/veupath/redmine/client/rnaseq.py [hunk 15]
```diff
@@ -13,10 +13,13 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""Redmine issue object for RNA-Seq datatype."""
 
 import re
-from unidecode import unidecode
 from typing import Any, Dict, List
+
+from unidecode import unidecode
```
Comment (JAlvarezJarreta) [inline src/veupath/redmine/client/rnaseq.py:21]: I may be wrong, but this is not used anywhere, is it?
```suggestion
```

File: pyproject.toml [hunk 16]
```diff
@@ -44,11 +44,14 @@ classifiers = [
 dependencies = [
     "biopython",
     "python-redmine",
+    "unidecode",
```
Comment (JAlvarezJarreta) [inline pyproject.toml:47]:
```suggestion
```
Not sure this import is needed (see below).

**Review by JAlvarezJarreta — CHANGES_REQUESTED** [review]
Great update. Couple of suggestions to homogenise the docstring content.

A side note comment: since you are updating the code, should we also consider moving from `os.path.join()` to start using `pathlib.Path()`?

File: src/veupath/redmine/client/rnaseq.py (hunk 15 above)
Comment (MatBarba) [reply src/veupath/redmine/client/rnaseq.py:21]: It's used in the `_normalize_name` method

File: pyproject.toml (hunk 16 above)
Comment (MatBarba) [reply pyproject.toml:47]: It is used in the rnaseq.py module

**Comment by MatBarba** [comment]
> Great update. Couple of suggestions to homogenise the docstring content.
> 
> A side note comment: since you are updating the code, should we also consider moving from `os.path.join()` to start using `pathlib.Path()`?

This PR is not so much an update of the code as making sure the code pass the base linters. Any other improvements should be made in separate PRs

**Review by JAlvarezJarreta — APPROVED** [review]

**Pull request merged**